### PR TITLE
Change DNS policy for vsphere-csi-node to default

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/area control-plane
/kind impediment
/platform vsphere

**What this PR does / why we need it**:
With the introduction of the feature [Wait for node-critical pods before scheduling workload pods](https://github.com/gardener/gardener/issues/7117) a deadlock was introduced for the vsphere-csi-node startup on new clusters.
The `vsphere-csi-node` tries to get the config map `internal-feature-states.csi.vsphere.vmware.com` from the kube-apiserver, but the DNS resolution fails, as the CoreDNS pods are still pending. They are waiting for the node taint `node.gardener.cloud/critical-components-not-ready` with effect `NoSchedule` is removed. But as the vsphere-csi-node is not ready, the taint is not removed.
The easy solution to resolve the deadlock is to change the DNS policy to "Default". As the vsphere-csi-node is running in the host network, the DNS resolution of the kube-apiserver name is performed by the node then.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Just for reference, here the error message in the vsphere-csi-node:
```txt
{"level":"error","time":"2023-03-29T15:39:37.394823348Z","caller":"k8sorchestrator/k8sorchestrator.go:364","msg":"failed to fetch configmap internal-feature-states.csi.vsphere.vmware.com from namespace kube-system. Error: Get \"https://api.bmahjo3yg1.xxx.internal.example.com:443/api/v1/namespaces/kube-system/configmaps/internal-feature-states.csi.vsphere.vmware.com\": dial tcp: lookup api.bmahjo3yg1.ixxx.internal.example.com on 100.104.0.10:53: write udp 100.104.0.10:51734->100.104.0.10:53: write: operation not permitted","TraceId":"0d9eca11-2214-49be-9bb6-f1e0802e2e14","TraceId":"1c0b19db-c25f-409e-9c47-fe3d67d716d7","stacktrace":"sigs.k8s.io/vsphere-csi-driver/v2/pkg/csi/service/common/commonco/k8sorchestrator.initFSS\n\t/build/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go:364\nsigs.k8s.io/vsphere-csi-driver/v2/pkg/csi/service/common/commonco/k8sorchestrator.Newk8sOrchestrator\n\t/build/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go:265\nsigs.k8s.io/vsphere-csi-driver/v2/pkg/csi/service/common/commonco.GetContainerOrchestratorInterface\n\t/build/pkg/csi/service/common/commonco/coagnostic.go:81\nsigs.k8s.io/vsphere-csi-driver/v2/pkg/csi/service.(*vsphereCSIDriver).BeforeServe\n\t/build/pkg/csi/service/driver.go:114\nsigs.k8s.io/vsphere-csi-driver/v2/pkg/csi/service.(*vsphereCSIDriver).Run\n\t/build/pkg/csi/service/driver.go:151\nmain.main\n\t/build/cmd/vsphere-csi/main.go:71\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
```

**Release note**:

```other operator
Change DNS policy for vsphere-csi-node to default
```

### CODE REVIEW

**Review by ScheererJ — COMMENTED** [review]
Would it make sense to explicitly set the kubernetes service host environment variable to ensure that csi-node always takes the external route?
I would assume that this now implicitly relies on the mutating webhook of apiserver-proxy to inject the kubernetes service host. However, we have seen scenarios, e.g. when `kube-apiserver` is shutting down, when this webhook fails, i.e. the pod will not get kubernetes service host injected. In that scenario csi-node might try to resolve `kube-apiserver` as `kubernetes.default.svc.cluster.local`, which will not work with `dnsPolicy: Default`.

Long story short: I would suggest to also specify kubernetes service host environment variable to ensure also edge cases will work.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Review by briantopping — APPROVED** [review]
Thanks Martin! :) 
/lgtm
